cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/etl/batch/sink/DBSinkTest.java

```diff
@@ -0,0 +1,162 @@
+import co.cask.cdap.etl.common.DBRecord;
+import co.cask.cdap.etl.common.MockEmitter;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import org.apache.hadoop.io.NullWritable;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ *
+ */
+public class DBSinkTest {
+
+  @Test
+  public void testIdentity() throws Exception {
+    final DBSink.DBSinkConfig config = new DBSink.DBSinkConfig();
+    config.columns = "ts,body";
+    config.tableName = "foo";
+
+    DBSink sink = new DBSink(config) {
+      @Override
+      public List<String> getColumns() {
+        return ImmutableList.copyOf(Splitter.on(",").split(config.columns));
+      }
+    };
+
+    StructuredRecord input = StructuredRecord
+      .builder(Schema.recordOf(
+        "foo",
+        Schema.Field.of("ts", Schema.of(Schema.Type.LONG)),
+        Schema.Field.of("body", Schema.of(Schema.Type.STRING))))
+      .set("ts", 123)
+      .set("body", "sdfsdf")
+      .build();
+
+    MockEmitter<KeyValue<DBRecord, NullWritable>> emitter = new MockEmitter<>();
+    sink.transform(input, emitter);
+
+    Assert.assertEquals(1, emitter.getEmitted().size());
+    Assert.assertEquals(
+      getRecordFields(input),
+      getRecordFields(emitter.getEmitted().get(0).getKey().getRecord()));
+  }
+
+  @Test
+  public void testDropColumn() throws Exception {
+    final DBSink.DBSinkConfig config = new DBSink.DBSinkConfig();
+    config.columns = "body,ts";
+    config.tableName = "foo";
+
+    DBSink sink = new DBSink(config) {
+      @Override
+      public List<String> getColumns() {
+        return ImmutableList.copyOf(Splitter.on(",").split(config.columns));
+      }
+    };
+
+    StructuredRecord input = StructuredRecord
+      .builder(Schema.recordOf(
+        "foo",
+        Schema.Field.of("ts", Schema.of(Schema.Type.LONG)),
+        Schema.Field.of("headers", Schema.mapOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.STRING))),
+        Schema.Field.of("body", Schema.of(Schema.Type.STRING))))
+      .set("ts", 123)
+      .set("headers", ImmutableMap.of("hello", "zzz"))
+      .set("body", "sdfsdf")
+      .build();
+
+    StructuredRecord output = StructuredRecord
+      .builder(Schema.recordOf(
+        "foo",
+        Schema.Field.of("body", Schema.of(Schema.Type.STRING)),
+        Schema.Field.of("ts", Schema.of(Schema.Type.LONG))))
+      .set("body", "sdfsdf")
+      .set("ts", 123)
+      .build();
+
+    MockEmitter<KeyValue<DBRecord, NullWritable>> emitter = new MockEmitter<>();
+    sink.transform(input, emitter);
+
+    Assert.assertEquals(1, emitter.getEmitted().size());
+    Assert.assertEquals(
+      getRecordFields(output),
+      getRecordFields(emitter.getEmitted().get(0).getKey().getRecord()));
+  }
+
+  @Test(expected = NullPointerException.class)
```

Comment:
nit: Where will this be thrown? If you are expecting a generic exception like ``NullPointerException`` (which could be thrown at multiple places, and not a specific exception, then it would be better to enclose the exact lines that throw ``NullPointerException`` in a try catch and ignore the exception.